app/src/main/java/com/example/instasnap/View/HomePage/PostRecyclerViewAdapter.java

```diff
@@ -11,18 +11,22 @@
 import androidx.recyclerview.widget.RecyclerView;
 
 import com.example.instasnap.Model.Post;
+import com.example.instasnap.Model.User;
 import com.example.instasnap.R;
+import com.example.instasnap.Utils.Parser;
 import com.example.instasnap.ViewModel.HomePageViewModel;
 
 import java.util.ArrayList;
 
 public class PostRecyclerViewAdapter extends RecyclerView.Adapter<PostRecyclerViewAdapter.PostViewHolder>{
+    private ArrayList<User> _users;
 
     private ArrayList<Post> _posts;
     private HomePageViewModel _homeHomePageViewModel;
 
-    public PostRecyclerViewAdapter(ArrayList<Post> posts, HomePageViewModel homePageViewModel){
-        _posts = posts;
+    public PostRecyclerViewAdapter(ArrayList<User> users, HomePageViewModel homePageViewModel){
+        _users = users;
+        _posts = Parser.getAllPosts(_users);
```

Comment:
As before if you pass here a reference of _allPosts you wont have to parse again and changes will apply everywhere.